chain-abstraction/sol-implementation/assertion_chain.go

```diff
@@ -105,8 +106,15 @@ func NewAssertionChain(
 	if err != nil {
 		return nil, err
 	}
+	challengeManagerAddr, err := assertionChainBinding.RollupUserLogicCaller.ChallengeManager(
```

Comment:
This is not exactly right. A challenge manager address is fixed per-challenge, but could change between challenges. That is, any time we do an operation in this package, we need to make sure it is at the correct challenge manager address. These are namespaced by assertion hash, so perhaps we can have a mapping

Reply:
Resolving based on offline discussion.